crates/packager/src/package/pacman/mod.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2024-2024 CrabNebula Ltd.
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use super::{
+    deb::{copy_custom_files, generate_data, tar_and_gzip_dir},
+    Context,
+};
+use crate::{config::Config, util};
+use heck::AsKebabCase;
+use sha2::{Digest, Sha512};
+use std::{
+    fs::File,
+    io::{self, Write},
+    path::{Path, PathBuf},
+};
+
+#[tracing::instrument(level = "trace")]
+pub(crate) fn package(ctx: &Context) -> crate::Result<Vec<PathBuf>> {
+    let Context {
+        config,
+        intermediates_path,
+        ..
+    } = ctx;
+
+    let arch = match config.target_arch()? {
+        "x86" => "i386",
+        "arm" => "armhf",
+        other => other,
+    };
+
+    let intermediates_path = intermediates_path.join("pacman");
+    util::create_clean_dir(&intermediates_path)?;
+
+    let package_base_name = format!("{}_{}_{}", config.main_binary_name()?, config.version, arch);
+    let package_name = format!("{}.tar.gz", package_base_name);
+
+    let pkg_dir = intermediates_path.join(&package_base_name);
+    let pkg_path = config.out_dir().join(&package_name);
+    let pkgbuild_path = pkg_path.with_file_name("PKGBUILD");
+
+    tracing::info!("Packaging {} ({})", package_name, pkg_path.display());
+
+    tracing::debug!("Generating data");
+    let _ = generate_data(config, &pkg_dir)?;
+
+    tracing::debug!("Copying files specified in `deb.files`");
+    if let Some(files) = config.deb().and_then(|d| d.files.as_ref()) {
```

Comment:
we shouldn't depend on `deb` config here, if a `files` option is needed, we should add it under `pacman` config